### PR TITLE
Allow the filter JSON to be passed inline in the /sync request

### PR DESCRIPTION
**Pull request by NegativeMjark**

### CODE REVIEW

File: api/client-server/sync.yaml [hunk 1]
```diff
@@ -32,7 +32,12 @@ paths:
           name: filter
           type: string
           description: |-
-            The ID of a filter created using the filter API.
+            The ID of a filter created using the filter API or a filter JSON
```
Comment (illicitonion) [inline api/client-server/sync.yaml]: Should the JSON be URL-encoded? Or something else?

Comment (NegativeMjark) [inline api/client-server/sync.yaml]: It should be passed like any other query parameter, encoded as UTF-8 and URL encoded.

Comment (illicitonion) [inline api/client-server/sync.yaml]: We do very explicitly state that all strings must be UTF-8 encoded in the API Standards section. As far as I know, this is the first place where a query parameter is anything other than a string or an int.

Comment (NegativeMjark) [inline api/client-server/sync.yaml]: Is saying that it is passed a string sufficient? c3a1318

**Comment by illicitonion** [comment]
LGTM

**Pull request merged**